#!/usr/bin/env python
PACKAGE = "exposure_control"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

# enable/disable exposure control
gen.add("enabled", bool_t,   0, "enable/disable auto exposure", True);
gen.add("min_gain", double_t, 0, "auto gain min", 0, 0, 10);
gen.add("max_gain", double_t, 0, "auto gain max", 0, 0, 10);
gen.add("min_shutter", double_t,0,"auto shutter min (msec)", 0.01, 0.001, 100);
gen.add("max_shutter", double_t, 0, "auto shutter max (msec)", 20, 1, 100);
gen.add("brightness", double_t, 0, "target brightness (0-255)", 70, 0, 255);
gen.add("brightness_tol", double_t, 0, "target brightness tolerance (0-255)", 30, 0, 255);
gen.add("auto_shutter_alpha", double_t, 0, "auto shutter alpha param ", 1.0, 0.1, 1.0);
gen.add("down_sampling_rate", int_t, 0, "down sampling rate (window size)", 32, 1, 64);
gen.add("top_margin", int_t, 0, "top margin (percent)", 30, 0, 100);
gen.add("bottom_margin", int_t, 0, "bottom margin (percent)", 0, 0, 100);
gen.add("wait_frames", int_t, 0, "frames to wait for change", 4, 0, 10);
exit(gen.generate(PACKAGE, "exposure_control", "ExposureControlDyn"))
